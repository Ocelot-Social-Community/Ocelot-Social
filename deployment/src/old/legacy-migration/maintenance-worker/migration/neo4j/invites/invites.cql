/*
// Alpha Model
// [ ] Not modeled in Nitro
// [X] Modeled in Nitro
// [-] Omitted in Nitro
// [?] Unclear / has work to be done for Nitro
    {
[ ]   email: {
[ ]     type: String,
[ ]     required: true,
[-]     index: true,
[ ]     unique: true
      },
[ ]   code: {
[ ]     type: String,
[-]     index: true,
[ ]     required: true
      },
[ ]   role: {
[ ]     type: String,
[ ]     enum: ['admin', 'moderator', 'manager', 'editor', 'user'],
[ ]     default: 'user'
      },
[ ]   invitedByUserId: { type: String },
[ ]   language: { type: String },
[ ]   badgeIds: [],
[ ]   wasUsed: {
[ ]     type: Boolean,
[-]     index: true
      },
[ ]   createdAt: {
[ ]     type: Date,
[ ]     default: Date.now
      },
[ ]   wasSeeded: { type: Boolean }
  }
*/

CALL apoc.load.json("file:${IMPORT_CHUNK_PATH_CQL_FILE}") YIELD value as invite;
