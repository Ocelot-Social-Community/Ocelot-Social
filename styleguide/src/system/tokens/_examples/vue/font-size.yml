#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
