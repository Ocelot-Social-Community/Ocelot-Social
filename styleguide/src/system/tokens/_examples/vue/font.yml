#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
