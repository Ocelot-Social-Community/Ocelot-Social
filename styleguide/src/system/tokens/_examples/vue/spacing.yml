#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
