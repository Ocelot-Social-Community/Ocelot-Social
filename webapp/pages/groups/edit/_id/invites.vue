<template>
  <div>
    <base-card>
      <invitation-list
        @generate-invite-code="generateGroupInviteCode"
        @invalidate-invite-code="invalidateInviteCode"
        :inviteCodes="group.inviteCodes"
        :copy-message="
          $t('invite-codes.invite-link-message-group', {
            groupName: group.name,
            network: $env.NETWORK_NAME,
          })
        "
      />
    </base-card>
  </div>
</template>

<script>
import InvitationList from '~/components/_new/features/Invitations/InvitationList.vue'
import { generateGroupInviteCode, invalidateInviteCode } from '~/graphql/InviteCode'

export default {
  components: {
    InvitationList,
  },
  props: {
    group: {
      type: Object,
      required: true,
    },
  },
  methods: {
    async generateGroupInviteCode(comment) {
      try {
        await this.$apollo.mutate({
          mutation: generateGroupInviteCode(),
          variables: {
            comment,
            groupId: this.group.id,
          },
          update: (_, { data: { generateGroupInviteCode } }) => {
            /*
            this.setCurrentUser({
              ...this.currentUser,
              inviteCodes: [...this.user.inviteCodes, generateGroupInviteCode],
            })
            */
          },
        })
        this.$toast.success(this.$t('invite-codes.create-success'))
      } catch (error) {
        this.$toast.error(this.$t('invite-codes.create-error', { error: error.message }))
      }
    },
    async invalidateInviteCode(code) {
      try {
        await this.$apollo.mutate({
          mutation: invalidateInviteCode(),
          variables: {
            code,
          },
          update: (_, { data: { _invalidateInviteCode } }) => {
            /*
            this.setCurrentUser({
              ...this.currentUser,
              inviteCodes: this.user.inviteCodes.map((inviteCode) => ({
                ...inviteCode,
                isValid: inviteCode.code === code ? false : inviteCode.isValid,
              })),
            })
            */
          },
        })
        this.$toast.success(this.$t('invite-codes.invalidate-success'))
      } catch (error) {
        this.$toast.error(this.$t('invite-codes.invalidate-error', { error: error.message }))
      }
    },
  },
}
</script>
