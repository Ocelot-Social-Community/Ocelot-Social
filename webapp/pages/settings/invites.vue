<template>
  <base-card>
    <h2 class="title">{{ $t('settings.invites.name') }}</h2>
    <empty icon="tasks" message="Coming Soonâ€¦" />
  </base-card>
</template>

<script>
import Empty from '~/components/Empty/Empty'

export default {
  components: {
    Empty,
  },
}
</script>
