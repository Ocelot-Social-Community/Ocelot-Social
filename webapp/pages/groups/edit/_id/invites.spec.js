import { render, screen, fireEvent } from '@testing-library/vue'

import invites from './invites.vue'

const localVue = global.localVue

describe('invites.vue', () => {
  let wrapper
  let mocks

  beforeEach(() => {
    mocks = {
      $t: jest.fn((v) => v),
      $apollo: {
        mutate: jest.fn(),
      },
      $env: {
        NETWORK_NAME: 'test-network',
        INVITE_LINK_LIMIT: 5,
      },
      $toast: {
        success: jest.fn(),
        error: jest.fn(),
      },
      localVue,
    }
  })

  const Wrapper = () => {
    return render(invites, {
      localVue,
      propsData: {
        group: {
          id: 'group1',
          name: 'Group 1',
          inviteCodes: [
            {
              code: 'INVITE1',
              comment: 'Test invite 1',
              redeemedByCount: 0,
              isValid: true,
            },
            {
              code: 'INVITE2',
              comment: 'Test invite 2',
              redeemedByCount: 1,
              isValid: false,
            },
          ],
        },
      },
      mocks,
      stubs: {
        'client-only': true,
      },
    })
  }

  it('renders', () => {
    wrapper = Wrapper()
    expect(wrapper.container).toMatchSnapshot()
  })

  describe('when a new invite code is generated', () => {
    beforeEach(async () => {
      wrapper = Wrapper()
      const createButton = screen.getByLabelText('invite-codes.generate-code')
      await fireEvent.click(createButton)
    })

    it('calls the mutation to generate a new invite code', () => {
      expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
        mutation: expect.anything(),
        update: expect.anything(),
        variables: {
          groupId: 'group1',
          comment: '',
        },
      })
    })

    it('shows a success message', () => {
      expect(mocks.$toast.success).toHaveBeenCalledWith('invite-codes.create-success')
    })
  })
})
