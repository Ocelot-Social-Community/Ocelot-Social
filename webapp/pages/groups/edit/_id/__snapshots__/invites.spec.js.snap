// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invites.vue renders 1`] = `
<div>
  <div>
    <article
      class="base-card"
    >
      <h3
        class="ds-heading ds-heading-h3"
      >
        invite-codes.group-invite-links
      </h3>
       
      <div
        class="ds-space"
        style="margin-top: 32px; margin-bottom: 32px;"
      />
       
      <div
        class="invitation-list"
      >
        <ul>
          <client-only-stub>
            <li
              class="invitation"
            >
              <div
                class="column1"
              >
                <div
                  class="code"
                >
                  
      INVITE1
      
                  <span
                    class="mdash"
                  >
                    â€”
                  </span>
                   
                  <span
                    class="comment"
                  >
                    Test invite 1
                  </span>
                </div>
                 
                <div>
                  <span>
                    
        invite-codes.redeemed-count-0
      
                  </span>
                </div>
              </div>
               
              <div
                class="actions"
              >
                <button
                  aria-label="invite-codes.copy-code"
                  class="copy-button base-button --icon-only --circle"
                  disabled="disabled"
                  type="button"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                   
                  <!---->
                   
                </button>
                 
                <button
                  aria-label="invite-codes.invalidate"
                  class="invalidate-button base-button --icon-only --circle"
                  type="button"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                   
                  <!---->
                   
                </button>
              </div>
            </li>
          </client-only-stub>
        </ul>
         
        <div
          class="create-invitation create-invitation"
        >
          <div>
            invite-codes.generate-code-explanation
          </div>
           
          <form
            class="generate-invite-code-form"
          >
            <div
              class="ds-form-item ds-input-size-base"
            >
              <label
                class="ds-input-label"
                style="display: none;"
              >
                
  

              </label>
              <div
                class="ds-input-wrap"
              >
                <!---->
                <input
                  class="ds-input"
                  name="comment"
                  placeholder="invite-codes.comment-placeholder"
                  tabindex="0"
                  type="text"
                />
                <!---->
              </div>
              <transition-stub
                name="ds-input-error"
              >
                <div
                  class="ds-input-error"
                  style="display: none;"
                >
                  
    
  
                </div>
              </transition-stub>
            </div>
             
            <button
              aria-label="invite-codes.generate-code"
              class="generate-invite-code base-button --icon-only --circle"
              type="submit"
            >
              <span
                class="base-icon"
              >
                <!---->
              </span>
               
              <!---->
               
            </button>
          </form>
        </div>
      </div>
       
      <!---->
    </article>
  </div>
</div>
`;
